python/test/t_ProjectionStrategy_std.py

```diff
@@ -0,0 +1,112 @@
+#! /usr/bin/env python
+
+import openturns as ot
+from openturns.usecases import flood_model
+
+ot.TESTPREAMBLE()
+
+ot.Log.Show(ot.Log.NONE)
+
+sampleSize = 500
+totalDegree = 7
+
+#
+print("+ Compute flood model by regression")
+fm = flood_model.FloodModel()
+inputDescription = fm.model.getInputDescription()
+marginals = [fm.distribution.getMarginal(i) for i in range(fm.dim)]
+basis = ot.OrthogonalProductPolynomialFactory(marginals)
+inputSample = fm.distribution.getSample(sampleSize)
+outputSample = fm.model(inputSample)
+selectionAlgorithm = ot.LeastSquaresMetaModelSelectionFactory()
+leastSquaresStrategy = ot.LeastSquaresStrategy(selectionAlgorithm)
+print("leastSquaresStrategy")
+print(leastSquaresStrategy)
+print("leastSquaresStrategy (repr)")
+print(leastSquaresStrategy.__repr__())
+print("leastSquaresStrategy (html)")
+print(leastSquaresStrategy._repr_html_())
+
+enumerateFunction = basis.getEnumerateFunction()
+basisSize = enumerateFunction.getBasisSizeFromTotalDegree(totalDegree)
+adaptiveStrategy = ot.FixedStrategy(basis, basisSize)
+chaosAlgo = ot.FunctionalChaosAlgorithm(
+    inputSample, outputSample, fm.distribution, adaptiveStrategy, leastSquaresStrategy
+)
+projectionStrategy = chaosAlgo.getProjectionStrategy()
+print("projectionStrategy")
+print(projectionStrategy)
+print("projectionStrategy (repr)")
+print(projectionStrategy.__repr__())
+print("projectionStrategy (html)")
+print(projectionStrategy._repr_html_())
+
+
+#
+print("+ Compute flood model by integration")
+integrationStrategy = ot.IntegrationStrategy()
+print("integrationStrategy")
+print(integrationStrategy)
+print("integrationStrategy (repr)")
+print(integrationStrategy.__repr__())
+print("integrationStrategy (html)")
+print(integrationStrategy._repr_html_())
+chaosAlgo = ot.FunctionalChaosAlgorithm(
+    inputSample, outputSample, fm.distribution, adaptiveStrategy, integrationStrategy
+)
+projectionStrategy = chaosAlgo.getProjectionStrategy()
+print("projectionStrategy")
+print(projectionStrategy)
+print("projectionStrategy (repr)")
+print(projectionStrategy.__repr__())
+print("projectionStrategy (html)")
+print(projectionStrategy._repr_html_())
+
+
+class RepeatedFloodOutputDimensionFunction(ot.OpenTURNSPythonFunction):
+    def __init__(self, outputDimension):
+        super().__init__(4, outputDimension)
+        self.fm = flood_model.FloodModel()
+        self.outputDimension = outputDimension
+
+    def _exec(self, x):
+        y = ot.Point(self.outputDimension)
+        y_base = fm.model(x)
+        for i in range(self.outputDimension):
+            y[i] = i + y_base[0]
+        return y
```

Comment:
a PythonFunction would have been enough here as no data is required

Reply:
Fixed at beaf844144800c7d9702e357d896dde12f2ee449